web/src/components/organisms/Project/ModelsMenu/hooks.ts

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { Model } from "@reearth-cms/components/molecules/Schema/types";
+import {
+  useGetModelsQuery,
+  useCreateModelMutation,
+  useCheckModelKeyAvailabilityLazyQuery,
+  Model as GQLModel,
+} from "@reearth-cms/gql/graphql-client-api";
+
+type Params = {
+  projectId?: string;
+  modelId?: string;
+};
+
+export default ({ projectId, modelId }: Params) => {
+  const [modelModalShown, setModelModalShown] = useState(false);
+  const [isKeyAvailable, setIsKeyAvailable] = useState(false);
+  const [CheckModelKeyAvailability, { data: keyData }] = useCheckModelKeyAvailabilityLazyQuery({
+    fetchPolicy: "no-cache",
+  });
+
+  const handleModelKeyCheck = useCallback(
+    async (projectId: string, key: string) => {
+      if (!projectId || !key) return false;
+      const response = await CheckModelKeyAvailability({ variables: { projectId, key } });
+      return response.data ? response.data.checkModelKeyAvailability.available : false;
+    },
+    [CheckModelKeyAvailability],
+  );
+
+  useEffect(() => {
+    setIsKeyAvailable(keyData?.checkModelKeyAvailability.available ?? false);
+  }, [keyData?.checkModelKeyAvailability]);
+
+  const { data } = useGetModelsQuery({
+    variables: { projectId: projectId ?? "", first: 100 },
+    skip: !projectId,
+  });
+
```

Comment:
Is pagination not setup yet on the backend?

Reply:
@KaWaite , not yet, and they made it like this.
Pagination would be a different task, as for the speed of development we've been asked to consider a small number of data, especially for models which will be a small number.